tests/test_versioning.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from packaging.specifiers import Specifier
+from packaging.version import Version
+
+from vvm import detect_vyper_version_from_source
+from vvm.exceptions import UnexpectedVersionError
+from vvm.utils.versioning import _detect_version_specifier, _pick_vyper_version
+
+LAST_PER_MINOR = {
+    1: Version("0.1.0b17"),
+    2: Version("0.2.16"),
+    3: Version("0.3.10"),
+}
+
+
+def test_foo_vyper_version(foo_source, all_versions):
+    specifier = _detect_version_specifier(foo_source)
+    assert str(specifier) == f"=={all_versions}"
+    assert all_versions.major == 0
+    assert _pick_vyper_version(specifier) == all_versions
+
+
+@pytest.mark.parametrize(
+    "version_str,decorator,pragma,expected_specifier,expected_version",
+    [
+        ("^0.1.1", "public", "@version", "~=0.1", "latest"),
```

Comment:
Something might be wrong here.
According to [the documentation](https://peps.python.org/pep-0440/#compatible-release), `~=0.1` should match `0.1.x` not the latest!

Reply:
My mistake, `~=0.1` matches `0.x`
Which is the same as the npm `^` which allows minor upgrades (major still `v0`).